ractor.c

```diff
@@ -1942,6 +1942,7 @@ move_enter(VALUE obj, struct obj_traverse_replace_data *data)
         VALUE type = RB_BUILTIN_TYPE(obj);
         type |= wb_protected_types[type] ? FL_WB_PROTECTED : 0;
         NEWOBJ_OF(moved, struct RBasic, 0, type, rb_gc_obj_slot_size(obj), 0);
+        MEMZERO((char *)moved + sizeof(VALUE), char, rb_gc_obj_slot_size(obj) - sizeof(VALUE));
```

Comment:
`klass` looks initialized as well as `flags`.

```suggestion
        size_t slot_size = rb_gc_obj_slot_size(obj);
        type |= wb_protected_types[type] ? FL_WB_PROTECTED : 0;
        NEWOBJ_OF(moved, struct RBasic, 0, type, slot_size, 0);
        MEMZERO(&moved[1], char, slot_size - sizeof(*moved));
```

Reply:
Good catch, thanks!